plugins/www/Optimove.js

```diff
@@ -1,44 +1,188 @@
+var exec = require("cordova/exec");
 
+var inAppInboxUpdatedHandler;
 
-var exec = require('cordova/exec');
+var currentConfig = {
+  pushReceivedHandler: pushReceivedHandler, //function that receives one argument, a push message object
+  pushOpenedHandler: pushOpenedHandler, //function that receives one argument, a push message object
+  inAppDeepLinkHandler: inAppDeepLinkHandler, //expect to be a function that receives one argument, a deepLink data object
+};
 
+function nativeMessageHandler(message) {
+  alert(message);
+  if (!message || typeof message === "string") {
+    return;
+  }
+  
+  const handlerName = `${message.type}Handler`;
+  alert(handlerName);
+  alert(typeof inAppInboxUpdatedHandler);
+  if (
+    handlerName === "inAppInboxUpdatedHandler" &&
+    typeof inAppInboxUpdatedHandler === "function"
+  ) {
+    inAppInboxUpdatedHandler();
+    return;
+  }
 
+  if (typeof currentConfig[handlerName] === "function") {
+    currentConfig[handlerName](message.data);
+  } else {
+    console.log(`Optimove: No handler defined for '${message.type}' event`);
+  }
+}
 
 const Optimove = {
-    setUserId: function (userId) {
+  initBaseSdk: function () {
```

Comment:
I'm not sure this is exposed on the public JS interface, I think it only needs called when the developer sets their callbacks for push/in-app etc.

Reply:
so should I create a js object that will represent the configuration of the optimove sdk, and add the handlers and this function also to that object?

Reply:
havent looked much at what does what, but is this just renaming? Base sdk is our OptimoveAndroid, so, `initBaseSdk` shall be an internal method. But could have `initBaseSdk` -> `initialize`?

Reply:
havent looked much at what does what, but is this just renaming? Base sdk is our OptimoveAndroid, so, `initBaseSdk` shall be an internal method. Could have `initBaseSdk` -> `initialize`?

Do we cuttently set push/in-app handlers?